src/main/java/ch/njol/skript/effects/EffShoot.java

```diff
@@ -40,99 +40,100 @@
 import ch.njol.skript.util.Direction;
 import ch.njol.util.Kleenean;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Shoot")
 @Description("Shoots a projectile (or any other entity) from a given entity.")
-@Examples({"shoot an arrow",
-		"make the player shoot a creeper at speed 10",
-		"shoot a pig from the creeper"})
+@Examples({
+	"shoot an arrow",
+	"make the player shoot a creeper at speed 10",
+	"shoot a pig from the creeper"
+})
 @Since("1.4")
 public class EffShoot extends Effect {
+
 	static {
 		Skript.registerEffect(EffShoot.class,
 				"shoot %entitydatas% [from %livingentities/locations%] [(at|with) (speed|velocity) %-number%] [%-direction%]",
 				"(make|let) %livingentities/locations% shoot %entitydatas% [(at|with) (speed|velocity) %-number%] [%-direction%]");
 	}
-	
-	private final static Double DEFAULT_SPEED = 5.;
-	
-	@SuppressWarnings("null")
+
+	private final static double DEFAULT_SPEED = 5.;
+
 	private Expression<EntityData<?>> types;
-	@SuppressWarnings("null")
 	private Expression<?> shooters;
-	@Nullable
-	private Expression<Number> velocity;
+
 	@Nullable
 	private Expression<Direction> direction;
-	
+
 	@Nullable
-	public static Entity lastSpawned = null;
-	
-	@SuppressWarnings({"unchecked", "null"})
+	private Expression<Number> velocity;
+
+	@Nullable
+	public static Entity lastSpawned;
+
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		types = (Expression<EntityData<?>>) exprs[matchedPattern];
 		shooters = exprs[1 - matchedPattern];
 		velocity = (Expression<Number>) exprs[2];
 		direction = (Expression<Direction>) exprs[3];
 		return true;
 	}
-	
-	@SuppressWarnings("null")
+
 	@Override
-	protected void execute(final Event e) {
+	protected void execute(Event event) {
 		lastSpawned = null;
-		final Number v = velocity != null ? velocity.getSingle(e) : DEFAULT_SPEED;
-		if (v == null)
-			return;
-		final Direction dir = direction != null ? direction.getSingle(e) : Direction.IDENTITY;
-		if (dir == null)
-			return;
-		for (final Object shooter : shooters.getArray(e)) {
-			for (final EntityData<?> d : types.getArray(e)) {
+		double velocity = DEFAULT_SPEED;
```

Comment:
It seems like you could make this final with an if(not null & is present)/else check and avoid reassigning it to the same thing.